modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/toggle_controls.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import getLexiconIcon from './get_lexicon_icon';
+import {setSessionValue} from './session.es';
+
+const EDIT_CONTROLS_STATE = Liferay._editControlsState;
+
+const MAP_TOGGLE_STATE = {
+	false: {
+		cssClass: 'controls-hidden',
+		iconCssClass: 'hidden',
+		state: 'hidden',
+	},
+	true: {
+		cssClass: 'controls-visible',
+		iconCssClass: 'view',
+		state: 'visible',
+	},
+};
+
+export default function toggleControls(node) {
+	const body = document.body;
+
+	node = node._node || body;
+
+	const trigger = node.querySelector('.toggle-controls');
+
+	if (!trigger) {
+		return;
+	}
+
+	let controlsVisible = EDIT_CONTROLS_STATE === 'visible';
```

Comment:
Good thought, it was a suggestion mostly, because of how it's defined, I assumed it would be changed less than the JS would get loaded - https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/html/common/themes/top_js.jspf#L430-L439